src/Compilers/Core/Portable/Generated/Operations.xml.Generated.cs

```diff
@@ -867,6 +867,92 @@ internal sealed partial class LazyBinaryOperatorExpression : BaseBinaryOperatorE
         protected override IOperation RightOperandImpl => _lazyRightOperand.Value;
     }
 
+    /// <summary>
+    /// Represents an operation with two operands that produces a result with the same type as at least one of the operands.
+    /// </summary>
+    internal abstract partial class BaseTupleBinaryOperatorExpression : Operation, ITupleBinaryOperation
+    {
+        public BaseTupleBinaryOperatorExpression(BinaryOperatorKind operatorKind, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit)
+            : base(OperationKind.BinaryOperator, semanticModel, syntax, type, constantValue, isImplicit)
+        {
+            OperatorKind = operatorKind;
+        }
+        /// <summary>
+        /// Kind of binary operation.
+        /// </summary>
+        public BinaryOperatorKind OperatorKind { get; }
+        /// <summary>
+        /// Left operand.
+        /// </summary>
+        public IOperation LeftOperand => Operation.SetParentOperation(LeftOperandImpl, this);
+        /// <summary>
+        /// Right operand.
+        /// </summary>
+        public IOperation RightOperand => Operation.SetParentOperation(RightOperandImpl, this);
+
+        protected abstract IOperation LeftOperandImpl { get; }
+        protected abstract IOperation RightOperandImpl { get; }
+
+        public override IEnumerable<IOperation> Children
+        {
+            get
+            {
+                if (LeftOperand != null)
+                {
+                    yield return LeftOperand;
+                }
+                if (RightOperand != null)
+                {
+                    yield return RightOperand;
+                }
```

Comment:
Are the operands ever `null`?

Reply:
I don't think so. I'm following the model of `BaseBinaryOperatorExpression` since this file is supposed to be generated. `BaseBinaryOperatorExpression` also has those checks (and I think the operands can never be null there either).

---
In reply to: [173851139](https://github.com/dotnet/roslyn/pull/25374#discussion_r173851139) [](ancestors = 173851139)